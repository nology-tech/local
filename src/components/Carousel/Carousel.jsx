import "./Carousel.scss";

const Carousel = ({ componentToDisplay }) => {
  return <div className="carousel">{componentToDisplay}</div>;
};

export default Carousel;
