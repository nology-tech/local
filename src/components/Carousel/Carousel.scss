.carousel {
  display: flex;
  overflow-x: scroll;
}
