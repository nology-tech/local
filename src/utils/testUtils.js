import { jest } from "@jest/globals";
import { render } from "@testing-library/react";
import { BrowserRouter as Router } from "react-router-dom";
import UserContext from "../context/UserContext";
import mockUser from "../data/mockUser";

export const customRender = (
  ui,
  useRouting = true,
  contextOptions = {
    useContext: true,
    value: { user: mockUser, setUser: jest.fn() },
  }
) => {
  // wrap components in routing and context if requested
  const routingUiResult = useRouting && wrapWithRouting(ui);
  const contextUiResult =
    contextOptions.useContext &&
    wrapWithContext(routingUiResult, contextOptions.value);

  // use RTL's render function to return the test component
  return render(contextUiResult);
};

const wrapWithRouting = (ui) => {
  return <Router>{ui}</Router>;
};

const wrapWithContext = (ui, value) => {
  return <UserContext.Provider value={value}>{ui}</UserContext.Provider>;
};
