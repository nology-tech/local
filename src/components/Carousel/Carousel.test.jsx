import { customRender } from "../../utils/testUtils";
import Carousel from "../Carousel/Carousel";

it("should render the text on the display", () => {
  const { container } = customRender(<Carousel />);

  expect(container).toMatchSnapshot();
});
