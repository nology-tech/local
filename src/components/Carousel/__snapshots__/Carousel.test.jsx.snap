// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render the text on the display 1`] = `
<div>
  <div
    class="carousel"
  />
</div>
`;
